google/cloud/spanner_v1/database.py

```diff
@@ -726,6 +734,17 @@ def execute_pdml():
 
                 return result_set.stats.row_count_lower_bound
 
+            observability_options = getattr(self, "observability_options", {})
+            if isinstance(observability_options, dict):
+                observability_options["db_name"] = self.name
+            with trace_call(
+                "CloudSpanner.execute_partitioned_pdml",
```

Comment:
The problem i see here is with every retry there will be a new span parent `CloudSpanner.execute_partitioned_pdml` which does not indicate that a retry is happening.
Instead we should capture the retries with in only one parent span `CloudSpanner.execute_partitioned_pdml`.